org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/CodeActionHandlerUseWorkspaceEditTest.java

```diff
@@ -0,0 +1,253 @@
+import org.eclipse.lsp4j.jsonrpc.messages.Either;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+/**
+ * @author Gorkem Ercan
+ *
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class CodeActionHandlerUseWorkspaceEditTest extends AbstractCompilationUnitBasedTest {
+
+	@Mock
+	private JavaClientConnection connection;
+
+
+	@Override
+	@Before
+	public void setup() throws Exception{
+		importProjects("eclipse/hello");
+		project = WorkspaceHelper.getProject("hello");
+		wcOwner = new LanguageServerWorkingCopyOwner(connection);
+		server = new JDTLanguageServer(projectsManager, this.preferenceManager);
+	}
+
+	@Test
+	public void testCodeAction_removeUnusedImport() throws Exception{
+		ICompilationUnit unit = getWorkingCopy(
+				"src/java/Foo.java",
+				"import java.sql.*; \n" +
+						"public class Foo {\n"+
+						"	void foo() {\n"+
+						"	}\n"+
+				"}\n");
+
+		CodeActionParams params = new CodeActionParams();
+		params.setTextDocument(new TextDocumentIdentifier(JDTUtils.toURI(unit)));
+		final Range range = CodeActionUtil.getRange(unit, "java.sql");
+		params.setRange(range);
+		params.setContext(new CodeActionContext(Arrays.asList(getDiagnostic(Integer.toString(IProblem.UnusedImport), range))));
+		List<Either<Command, CodeAction>> codeActions = getCodeActions(params);
+		Assert.assertNotNull(codeActions);
+		Assert.assertTrue(codeActions.size() >= 3);
+		Assert.assertEquals(codeActions.get(0).getRight().getKind(), CodeActionKind.QuickFix);
+		Assert.assertEquals(codeActions.get(1).getRight().getKind(), CodeActionKind.QuickFix);
+		Assert.assertEquals(codeActions.get(2).getRight().getKind(), CodeActionKind.SourceOrganizeImports);
+		WorkspaceEdit w = codeActions.get(0).getRight().getEdit();
+		Assert.assertNotNull(w);
+	}
+
+	@Test
+	public void testCodeAction_removeUnterminatedString() throws Exception{
+		ICompilationUnit unit = getWorkingCopy(
+				"src/java/Foo.java",
+				"public class Foo {\n"+
+						"	void foo() {\n"+
+						"String s = \"some str\n" +
+						"	}\n"+
+				"}\n");
+
+		CodeActionParams params = new CodeActionParams();
+		params.setTextDocument(new TextDocumentIdentifier(JDTUtils.toURI(unit)));
+		final Range range = CodeActionUtil.getRange(unit, "some str");
+		params.setRange(range);
+		params.setContext(new CodeActionContext(Arrays.asList(getDiagnostic(Integer.toString(IProblem.UnterminatedString), range))));
+		List<Either<Command, CodeAction>> codeActions = getCodeActions(params);
+		Assert.assertNotNull(codeActions);
+		Assert.assertFalse(codeActions.isEmpty());
+		Assert.assertEquals(codeActions.get(0).getRight().getKind(), CodeActionKind.QuickFix);
+		WorkspaceEdit w = codeActions.get(0).getRight().getEdit();
+		Assert.assertNotNull(w);
+	}
+
+	@Test
+	public void testCodeAction_exception() throws JavaModelException {
```

Comment:
This test case is same as CodeActionHandlerTest. No need of it.

Reply:
Fixed in da69be95d3517518867f94a1e98817b2e7f39f0f.
(https://github.com/eclipse/eclipse.jdt.ls/commit/da69be95d3517518867f94a1e98817b2e7f39f0f#diff-40c092ea2f1a4bf03e8de7891cb6625eL125)